src/components/_global/BalChart/BalChart.vue

```diff
@@ -315,203 +205,228 @@ export default defineComponent({
+    },
+    lineStyle: {
+      width: 2
+    },
+    areaStyle: props.areaStyle,
+    itemStyle: {
+      borderRadius: 100
+    },
+    emphasis: {
+      itemStyle: {
+        color: props.hoverColor,
+        borderColor: props.hoverBorderColor
+      }
+    },
+    // This is a retrofitted option to show the small pill with the
+    // latest value of the series at the end of the line on the RHS
+    // the line is hidden, but the label is shown with extra styles
+    markLine: {
+      symbol: 'roundRect',
+      symbolSize: 0,
+      lineStyle: {
+        color: 'rgba(0, 0, 0, 0)'
       },
-      series: props.data.map((d, i) => ({
-        data: d.values,
-        type: props.chartType,
-        smooth: 0.3,
-        showSymbol: false,
-        name: d.name,
-        silent: true,
-        animationEasing: function(k) {
-          return k === 1 ? 1 : 1 - Math.pow(2, -10 * k);
-        },
-        lineStyle: {
-          width: 2
+      precision: 5,
+      label: {
+        backgroundColor: (props.color || [])[i] || 'black',
+        borderRadius: 3,
+        padding: 4,
+        formatter: (params: any) => {
+          return fNum2(params.data.yAxis, props.axisLabelFormatter.yAxis);
         },
-        areaStyle: props.areaStyle,
-        itemStyle: {
-          borderRadius: 100
-        },
-        emphasis: {
-          itemStyle: {
-            color: props.hoverColor,
-            borderColor: props.hoverBorderColor
-          }
-        },
-        // This is a retrofitted option to show the small pill with the
-        // latest value of the series at the end of the line on the RHS
-        // the line is hidden, but the label is shown with extra styles
-        markLine: {
-          symbol: 'roundRect',
-          symbolSize: 0,
-          lineStyle: {
-            color: 'rgba(0, 0, 0, 0)'
-          },
-          precision: 5,
-          label: {
-            backgroundColor: (props.color || [])[i] || 'black',
-            borderRadius: 3,
-            padding: 4,
-            formatter: (params: any) => {
-              return fNum2(params.data.yAxis, props.axisLabelFormatter.yAxis);
-            },
-            color: '#FFF',
-            fontSize: 10
-          },
-          data: props.isLastValueChipVisible
-            ? [
-                {
-                  name: 'Latest',
-                  yAxis: (last(props.data[i]?.values) || [])[1]
-                }
-              ]
-            : [],
-          animation: false
-        }
-      }))
-    }));
-
-    const styleOverrides = computed(() => {
-      let style: any = {};
-      if (props.height && typeof props.height === 'number') {
-        style.height = `${props.height}px`;
-      }
-      return style;
-    });
+        color: '#FFF',
+        fontSize: 10
+      },
+      data: props.isLastValueChipVisible
+        ? [
+            {
+              name: 'Latest',
+              yAxis: (last(props.data[i]?.values) || [])[1]
```

Comment:
This will error if props.data[i] is an empty array or null, as it'll try and read the 2nd value of an empty array. It does look like this was in the old code though, so maybe it's not possible for props.data[i] to be empty.

Reply:
yeah, seems that it's not possible for props.data[i] to be empty. nevertheless, fixed it